# 안녕하세요^^ 
## AIVLE 1차 미니프로젝트 '서울시 생활정보 기반 대중교통 수요 분석 ' 과정에 오신 여러분을 환영합니다.
* 본 과정에서는 실제 사례와 데이터를 기반으로 문제를 해결하는 전체 과정을 자기 주도형 실습으로 진행해볼 예정입니다.
* 앞선 교육과정을 정리하는 마음과 지금까지 배운 내용을 바탕으로 문제 해결을 해볼게요!
* 미니 프로젝트를 통한 문제 해결 과정 'A에서 Z까지', 지금부터 시작합니다!

---

# 데이터 분석부터 먼저 시작해보겠습니다.
## "구 별 유동인구 데이터" 를 확인해 보도록 하겠습니다

In [1]:
# 필요 라이브러리부터 설치합니다.
%pip install pandas seaborn

Note: you may need to restart the kernel to use updated packages.


## 기본전제
* 처음에 제공되는 기본/추가 데이터는 'raw_data' 폴더에 있습니다.

[기본 데이터]
* 1.2 seoul_moving_month_4.csv

[데이터 소개]
* 서울 시 구별 이동 2022년 4월 데이터

[변수 소개]
* 도착시간 : 0 -> "0:00~0:59"를 의미 
* 출발 자치구 : 자치구 코드
* 성별 : F(여성), M(남성)
* 나이 : 10 ~ 79세 (5세 단위), 0 ~ 10세, 80세 이상으로 구분
* 유형 : H(야간상주지), W(주간상주지), E(기타)
* 평균 이동 시간 : 동일 열 내 이동인구의 월 단위 평균 이동 시간 (10분단위)
* 이동인구 : 추정 합산값으로 소수점 표출, 개인정보 비식별화를 위해 3명 미만의 경우 * 처리

---

# 1.데이터 불러오기
## 모든 미니 프로젝트의 시작은 '데이터 불러오기' 부터라고 할 수 있습니다.
+ KeyPoint : 불러오고자 하는 데이터에 따라 자유롭게 변수로 지정할 수 있다.

###  데이터 프레임을 불러오고 변수로 저장(여기서는 CSV 기준으로 진행)
* csv : pd.read_csv("파일이름. csv")
* txt : pd.read_csv("파일이름. csv", sep="구분자")
* xlsx : pd.read_excel('파일이름.xlsx')
* pickle : pd.read_pickle("파일이름.pkl") <br>
 [참고] pickle은 파이썬의 모든 객체를 파일로 저장할 수 있는 방법으로 DataFrame,List,Dict 등 모든 객체 저장 가능(특히 sklearn라이브러리를 통해 모델을 학습시키고, 저장할 때 많이 사용)

#### [실습문제1] 데이터 로딩
* Pandas 라이브러리를 활용해서 'seoul_moving_month_4_step1.csv' 데이터를 확인하세요.
    * 아래에 'Pandas'를 Import하고 'seoul_moving_month_4_step1.csv'파일을 'seoul_moving'에 불러오고,  Shape을 확인하는 소스코드를 작성하고 실행해보세요.
    * 데이터 파일 로딩시 참고 사항 
        * 구분자(sep)는 ',' 입니다
        * UTF-8 인코더를 사용해 주세요
        * index column은 0번 열 입니다

In [2]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
import pandas as pd

seoul_moving = pd.read_csv('1.2 seoul_moving_month_4.csv',sep=",", index_col = 0, encoding = 'UTF-8')

C:\Users\User\anaconda3\lib\site-packages\numpy\lib\arraysetops.py:583: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  mask |= (ar1 == a)


In [3]:
# 데이터 프레임의 Shape을 확인합니다.
seoul_moving.shape

(28923054, 10)

---

# 2.기본 정보 확인 및 클렌징

+ 데이터 클렌징 : 결측치, 이상치 등을 제거하여 데이터 분석 결과가 왜곡 되는 문제를 방지하기 위한 정제 과정

#### [실습문제2] 기본 정보 확인하기
* 'seoul_moving' 데이터의 정보를 확인해보세요.
* 'describe', 'info', 'head' 등 전부 활용해 보겠습니다.

In [4]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
seoul_moving.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 28923054 entries, 0 to 968462
Data columns (total 10 columns):
 #   Column       Dtype 
---  ------       ----- 
 0   대상연월         int64 
 1   요일           object
 2   도착시간         int64 
 3   출발 시군구 코드    int64 
 4   도착 시군구 코드    int64 
 5   성별           object
 6   나이           int64 
 7   이동유형         object
 8   평균 이동 시간(분)  int64 
 9   이동인구(합)      object
dtypes: int64(6), object(4)
memory usage: 2.4+ GB


In [6]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
seoul_moving.head(10)

대상연월 요일  도착시간  출발 시군구 코드  도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분) 이동인구(합)
0  202204  일     0      11010      11010  F   0   EE           10   28.02
1  202204  일     0      11010      11010  F   0   EH           10   27.52
2  202204  일     0      11010      11010  F   0   HE           10   27.83
3  202204  일     0      11010      11010  F  10   HW           10    6.05
4  202204  일     0      11010      11010  F  10   EH           90    5.95
5  202204  일     0      11010      11010  F  10   WH           10   17.89
6  202204  일     0      11010      11010  F  15   EH           10  114.23
7  202204  일     0      11010      11010  F  15   HH           20   20.07
8  202204  일     0      11010      11010  F  15   HE           10    8.03
9  202204  일     0      11010      11010  F  15   HW           10    8.04

In [8]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
seoul_moving.isna().sum()

대상연월           0
요일             0
도착시간           0
출발 시군구 코드      0
도착 시군구 코드      0
성별             0
나이             0
이동유형           0
평균 이동 시간(분)    0
이동인구(합)        0
dtype: int64

In [15]:
seoul_moving.describe()

대상연월          도착시간     출발 시군구 코드     도착 시군구 코드            나이  \
count  28923054.0  2.892305e+07  2.892305e+07  2.892305e+07  2.892305e+07   
mean     202204.0  1.365597e+01  1.712362e+04  1.719750e+04  4.279614e+01   
std           0.0  5.849675e+00  9.046478e+03  9.063818e+03  1.777485e+01   
min      202204.0  0.000000e+00  1.101000e+04  1.101000e+04  0.000000e+00   
25%      202204.0  1.000000e+01  1.110000e+04  1.111000e+04  3.000000e+01   
50%      202204.0  1.400000e+01  1.120000e+04  1.120000e+04  4.500000e+01   
75%      202204.0  1.800000e+01  3.101400e+04  3.102100e+04  5.500000e+01   
max      202204.0  2.300000e+01  3.900000e+04  3.900000e+04  8.000000e+01   

        평균 이동 시간(분)  
count  2.892305e+07  
mean   3.279829e+01  
std    2.584213e+01  
min    1.000000e+01  
25%    2.000000e+01  
50%    3.000000e+01  
75%    4.000000e+01  
max    7.700000e+02

* 1번 열의 의미
    * " 2022년 1월 일요일에 11010(종로구)에서 11010(종로구)으로 이동하여 0시 ~ 0시 59분 사이에 도착했고, E(기타)에서 H(야간상주지(집))으로 이동한 0 ~ 10세 여성은 총 27.52명이며 이들의 평균 이동시간은 10분이다

#### [실습문제3] 결측치 처리 ( NaN이 아닌 결측치를 처리 해봅시다)

* '인구이동(합)' column은 이동 인구의 합산인데 object 타입입니다. 

    - 왜 일까요? 

    - Tip! (이동인구 : 추정 합산값으로 소수점 표출, 개인정보 비식별화를 위해 3명 미만의 경우 * 처리)

    - 해당 값을 나름의 기준으로 변경하세요

    - '인구이동(합)' 컬럼의 데이터 타입을 수치형으로 변경하세요 

In [14]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
seoul_moving['이동인구(합)'].value_counts()

대상연월    요일  도착시간  출발 시군구 코드  도착 시군구 코드  성별  나이  이동유형  평균 이동 시간(분)  이동인구(합)
202204  금   0     11010      11010      F   0   EH    10           27.77      1
        일   18    11190      11240      M   20  EH    30           10.83      1
                                            25  WE    50           6.83       1
                                                HW    30           *          1
                                                HE    40           13.63      1
                                                                             ..
        수   10    11020      31023      M   25  WE    30           *          1
                                                HW    60           6.81       1
                                                HE    30           *          1
                                                EE    30           *          1
        화   23    39000      11250      M   60  EH    60           7.1        1
Length: 28923054, dtype: int64

In [17]:
# 최빈값과 평균값을 알 수 없는 상황이라 중위값으로 대체
seoul_moving['이동인구(합)'].replace("*", "2", inplace=True)
seoul_moving['이동인구(합)'].value_counts()

2          3842041
3.3         399374
6.05        366830
3.29        289731
3.31        232069
            ...   
2451.11          1
1870.7           1
1316.79          1
1141.72          1
1694.93          1
Name: 이동인구(합), Length: 135812, dtype: int64

In [ ]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.


In [19]:
# 이동인구(합) column의 데이터 타입을 바꿔주세요
seoul_moving = seoul_moving.astype({'이동인구(합)' : 'float'})
seoul_moving.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 28923054 entries, 0 to 968462
Data columns (total 10 columns):
 #   Column       Dtype  
---  ------       -----  
 0   대상연월         int64  
 1   요일           object 
 2   도착시간         int64  
 3   출발 시군구 코드    int64  
 4   도착 시군구 코드    int64  
 5   성별           object 
 6   나이           int64  
 7   이동유형         object 
 8   평균 이동 시간(분)  int64  
 9   이동인구(합)      float64
dtypes: float64(1), int64(6), object(3)
memory usage: 2.4+ GB


#### [실습문제4] 이동유형, 요일, 성별 변수의 value수를 파악

* 역시나 value_counts()를 활용해 봅시다.

In [22]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
seoul_moving['이동유형'].value_counts()

EH    5757073
HE    5520005
EE    5085099
WH    3108026
WE    2809148
HW    2695189
EW    2304535
WW    1026134
HH     617845
Name: 이동유형, dtype: int64

In [23]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
seoul_moving['요일'].value_counts()

토    4690806
금    4641881
목    4011291
수    3955646
화    3926876
월    3906272
일    3790282
Name: 요일, dtype: int64

In [24]:
seoul_moving['성별'].value_counts()

M    16355029
F    12568025
Name: 성별, dtype: int64

In [25]:
seoul_moving.head()

대상연월 요일  도착시간  출발 시군구 코드  도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분)  이동인구(합)
0  202204  일     0      11010      11010  F   0   EE           10    28.02
1  202204  일     0      11010      11010  F   0   EH           10    27.52
2  202204  일     0      11010      11010  F   0   HE           10    27.83
3  202204  일     0      11010      11010  F  10   HW           10     6.05
4  202204  일     0      11010      11010  F  10   EH           90     5.95

* 4월의 요일 수를 알려드릴게요
    * 일 : 4, 월 : 4, 화 : 4, 수 : 4, 목 : 4, 금 : 5, 토 : 5

In [ ]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.


#### [실습문제5] 어느 요일의 심야 이동이 가장 많을까요?

In [95]:
# 아래에 실습코드를 작성하고 결과를 확인합니다. (0시-4시59분까지)
seoul_moving_night = seoul_moving[seoul_moving['도착시간'] <=5 ]
seoul_moving_night['요일'].value_counts() # 불금

토    501891
금    435296
일    383611
목    377407
수    371376
화    359741
월    338716
Name: 요일, dtype: int64

#### [실습문제6] 서울 지역의 데이터만 골라 보도록 하겠습니다.

* 우리는 서울시의 데이터만 필요합니다!

* 출발과 도착이 모두 서울인 데이터만 추출해 주세요.

In [46]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# 서울은 다 10000으로 시작함, 경기도는 20000
only_seoul_moving = seoul_moving.loc[(seoul_moving['출발 시군구 코드'] < 20000) & (seoul_moving['도착 시군구 코드'] < 20000)]

In [47]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
only_seoul_moving.head()

대상연월 요일  도착시간  출발 시군구 코드  도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분)  이동인구(합)
0  202204  일     0      11010      11010  F   0   EE           10    28.02
1  202204  일     0      11010      11010  F   0   EH           10    27.52
2  202204  일     0      11010      11010  F   0   HE           10    27.83
3  202204  일     0      11010      11010  F  10   HW           10     6.05
4  202204  일     0      11010      11010  F  10   EH           90     5.95

#### [실습문제7] 서울지역의 구 별로 groupby하여 데이터를 확인하겠습니다 (유동인구 기준)

* 유출이 제일 많은 구는?

* 유입이 제일 많은 구는?

* 유출이 제일 적은 구는?

* 유입이 제일 적은 구는?

* 참고자료 서울 구별 코드
    * 11010	종로구
    * 11020	중구
    * 11030	용산구
    * 11040	성동구
    * 11050	광진구
    * 11060	동대문구
    * 11070	중랑구
    * 11080	성북구
    * 11090	강북구
    * 11100	도봉구
    * 11110	노원구
    * 11120	은평구
    * 11130	서대문구
    * 11140	마포구
    * 11150	양천구
    * 11160	강서구
    * 11170	구로구
    * 11180	금천구
    * 11190	영등포구
    * 11200	동작구
    * 11210	관악구
    * 11220	서초구
    * 11230	강남구
    * 11240	송파구
    * 11250	강동구

In [49]:
only_seoul_moving['출발 시군구 코드'] = only_seoul_moving['출발 시군구 코드'].map({
    11010: '종로구',
    11020: '중구',
    11030: '용산구',
    11040: '성동구',
    11050: '광진구',
    11060: '동대문구',
    11070: '중랑구',
    11080: '성북구',
    11090: '강북구',
    11100: '도봉구',
    11110: '노원구',
    11120: '은평구',
    11130: '서대문구',
    11140: '마포구',
    11150: '양천구',
    11160: '강서구',
    11170: '구로구',
    11180: '금천구',
    11190: '영등포구',
    11200: '동작구',
    11210: '관악구',
    11220: '서초구',
    11230: '강남구',
    11240: '송파구',
    11250: '강동구'})

<ipython-input-49-35f1534fa2e8>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  only_seoul_moving['출발 시군구 코드'] = only_seoul_moving['출발 시군구 코드'].map({


In [50]:
only_seoul_moving['도착 시군구 코드'] = only_seoul_moving['도착 시군구 코드'].map({
    11010: '종로구',
    11020: '중구',
    11030: '용산구',
    11040: '성동구',
    11050: '광진구',
    11060: '동대문구',
    11070: '중랑구',
    11080: '성북구',
    11090: '강북구',
    11100: '도봉구',
    11110: '노원구',
    11120: '은평구',
    11130: '서대문구',
    11140: '마포구',
    11150: '양천구',
    11160: '강서구',
    11170: '구로구',
    11180: '금천구',
    11190: '영등포구',
    11200: '동작구',
    11210: '관악구',
    11220: '서초구',
    11230: '강남구',
    11240: '송파구',
    11250: '강동구'})

<ipython-input-50-4b1f18d010c4>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  only_seoul_moving['도착 시군구 코드'] = only_seoul_moving['도착 시군구 코드'].map({


In [51]:
only_seoul_moving

대상연월 요일  도착시간 출발 시군구 코드 도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분)  이동인구(합)
0       202204  일     0       종로구       종로구  F   0   EE           10    28.02
1       202204  일     0       종로구       종로구  F   0   EH           10    27.52
2       202204  일     0       종로구       종로구  F   0   HE           10    27.83
3       202204  일     0       종로구       종로구  F  10   HW           10     6.05
4       202204  일     0       종로구       종로구  F  10   EH           90     5.95
...        ... ..   ...       ...       ... ..  ..  ...          ...      ...
920410  202204  토    23       강동구       강동구  M  80   WH           10    85.10
920411  202204  토    23       강동구       강동구  M  80   EH           10    28.36
920412  202204  토    23       강동구       강동구  M  80   EE           20    37.99
920413  202204  토    23       강동구       강동구  M  80   HE           10     9.42
920414  202204  토    23       강동구       강동구  M  80   WE           10     9.51

[10655895 rows x 10 columns]

In [98]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
pd.options.display.float_format = '{: .2f}'.format 
only_seoul_moving.groupby(by=['출발 시군구 코드'], as_index=False)[['이동인구(합)']].sum().sort_values('이동인구(합)', ascending=False)

출발 시군구 코드      이동인구(합)
0        강남구  30360975.84
17       송파구  23635229.17
14       서초구  19018210.06
19      영등포구  16155374.54
12       마포구  14853695.38
1        강동구  14491908.66
3        강서구  13981191.57
8        노원구  12512538.86
23        중구  11428793.40
16       성북구  11396450.73
4        관악구  11384453.84
22       종로구  11073367.84
5        광진구  10580143.60
13      서대문구  10317093.65
21       은평구  10300207.09
11       동작구  10214315.90
15       성동구  10177847.73
10      동대문구   9384812.46
6        구로구   9222969.32
18       양천구   8997185.03
24       중랑구   8359405.99
20       용산구   8243953.44
2        강북구   7603967.82
9        도봉구   6453345.46
7        금천구   5204412.28

In [99]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
pd.options.display.float_format = '{: .2f}'.format 
only_seoul_moving.groupby(by=['도착 시군구 코드'], as_index=False)[['이동인구(합)']].sum().sort_values('이동인구(합)', ascending=False)

도착 시군구 코드      이동인구(합)
0        강남구  29188819.52
17       송파구  23004946.79
14       서초구  18651195.64
19      영등포구  15693027.00
12       마포구  15083231.22
1        강동구  14404079.28
3        강서구  14016047.01
8        노원구  12538744.48
4        관악구  11638671.68
23        중구  11533349.41
22       종로구  11528599.19
16       성북구  11278789.10
5        광진구  11022136.66
21       은평구  10460584.26
15       성동구  10383741.91
13      서대문구  10203725.49
11       동작구  10119602.87
10      동대문구   9517475.46
18       양천구   9289534.65
6        구로구   9174396.80
20       용산구   8738356.25
24       중랑구   8477045.57
2        강북구   7735467.90
9        도봉구   6540855.44
7        금천구   5129426.08

In [ ]:
# 유입과 유출이 가장 많은 자치구는 강남구 , 가장 적은 구는 금천구

#### [실습문제8] 낮시간만 볼게요 심야버스 제외하겠습니다

* 도착시간 0부터 4까지(0:00 ~ 4:59) 제거

In [75]:
only_seoul_moving.loc[(only_seoul_moving['도착시간'] >= 0) & (only_seoul_moving['도착시간'] <= 4)]

대상연월 요일  도착시간 출발 시군구 코드 도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분)  이동인구(합)
0       202204  일     0       종로구       종로구  F   0   EE           10    28.02
1       202204  일     0       종로구       종로구  F   0   EH           10    27.52
2       202204  일     0       종로구       종로구  F   0   HE           10    27.83
3       202204  일     0       종로구       종로구  F  10   HW           10     6.05
4       202204  일     0       종로구       종로구  F  10   EH           90     5.95
...        ... ..   ...       ...       ... ..  ..  ...          ...      ...
307542  202204  토     4       강동구       강동구  M  80   WH           10    28.34
307543  202204  토     4       강동구       강동구  M  80   WE           10    85.68
307544  202204  토     4       강동구       강동구  M  80   EE           10    38.07
307545  202204  토     4       강동구       강동구  M  80   EH           10    28.36
307546  202204  토     4       강동구       강동구  M  80   WW           10    19.04

[1099957 rows x 10 columns]

In [77]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
only_seoul_moving = only_seoul_moving.drop(only_seoul_moving.loc[(only_seoul_moving['도착시간'] >= 0) & (only_seoul_moving['도착시간'] <= 4)].index, axis=0)

In [79]:
only_seoul_moving.reset_index()

index    대상연월 요일  도착시간 출발 시군구 코드 도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분)  \
0         14654  202204  일     5       노원구       성북구  F  65   EW           10   
1         14655  202204  일     5       노원구       성북구  F  65   WH           10   
2         14656  202204  일     5       노원구       성북구  F  70   HW           10   
3         14657  202204  일     5       노원구       성북구  F  70   HE           10   
4         14658  202204  일     5       노원구       성북구  F  70   EH           20   
...         ...     ... ..   ...       ...       ... ..  ..  ...          ...   
5489130  920410  202204  토    23       강동구       강동구  M  80   WH           10   
5489131  920411  202204  토    23       강동구       강동구  M  80   EH           10   
5489132  920412  202204  토    23       강동구       강동구  M  80   EE           20   
5489133  920413  202204  토    23       강동구       강동구  M  80   HE           10   
5489134  920414  202204  토    23       강동구       강동구  M  80   WE           10   

         이동인구(합)  
0           4.68  
1           9.44  
2           5.69  
3           5.68  
4           5.73  
...          ...  
5489130    85.10  
5489131    28.36  
5489132    37.99  
5489133     9.42  
5489134     9.51  

[5489135 rows x 11 columns]

#### [실습문제9] '도착 시군구 코드' 를 기준으로 각 구별로 '평균 이동 시간(분)' , '이동인구(합)' 의 평균 과 합을 구해보겠습니다

* groupby 사용, as_index=False로 설정해주세요

In [86]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
only_seoul_moving_mean = only_seoul_moving.groupby(by='도착 시군구 코드', as_index=False)[['평균 이동 시간(분)', '이동인구(합)']].mean()
only_seoul_moving_mean

도착 시군구 코드  평균 이동 시간(분)     이동인구(합)
0        강남구    23.418794  104.459927
1        강동구    25.931701   73.493575
2        강북구    24.355289   42.258539
3        강서구    25.520396   65.193949
4        관악구    23.480725   53.060305
5        광진구    22.810568   50.076721
6        구로구    23.162536   45.044959
7        금천구    24.283819   32.868927
8        노원구    24.662041   59.064504
9        도봉구    25.138206   40.239780
10      동대문구    22.149241   41.070868
11       동작구    22.059056   45.151199
12       마포구    22.307964   60.546047
13      서대문구    22.411314   46.169868
14       서초구    23.119681   72.540004
15       성동구    22.150334   44.749214
16       성북구    22.716321   48.965617
17       송파구    25.735699   94.967973
18       양천구    23.209394   49.181423
19      영등포구    21.745571   61.763388
20       용산구    21.820458   38.444154
21       은평구    24.993078   55.061792
22       종로구    21.867754   46.074772
23        중구    21.524817   46.243086
24       중랑구    23.887741   43.680575

In [100]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
pd.options.display.float_format = '{: .2f}'.format 
only_seoul_moving_sum = only_seoul_moving.groupby(by='도착 시군구 코드', as_index=False)[['평균 이동 시간(분)', '이동인구(합)']].sum()
only_seoul_moving_sum

도착 시군구 코드  평균 이동 시간(분)      이동인구(합)
0        강남구      6543820  29188819.52
1        강동구      5082380  14404079.28
2        강북구      4458260   7735467.90
3        강서구      5486630  14016047.01
4        관악구      5150450  11638671.68
5        광진구      5020720  11022136.66
6        구로구      4717560   9174396.80
7        금천구      3789660   5129426.08
8        노원구      5235480  12538744.48
9        도봉구      4086140   6540855.44
10      동대문구      5132710   9517475.46
11       동작구      4944030  10119602.87
12       마포구      5557360  15083231.22
13      서대문구      4952990  10203725.49
14       서초구      5944440  18651195.64
15       성동구      5139830  10383741.91
16       성북구      5232500  11278789.10
17       송파구      6234190  23004946.79
18       양천구      4383860   9289534.65
19      영등포구      5525180  15693027.00
20       용산구      4959790   8738356.25
21       은평구      4748160  10460584.26
22       종로구      5471640  11528599.19
23        중구      5368440  11533349.41
24       중랑구      4635870   8477045.57

In [102]:
# 파일을 합쳐볼까요
temp = pd.merge(only_seoul_moving,only_seoul_moving_sum,how='inner', on='도착 시군구 코드')
df_seoul_moving = pd.merge(temp,only_seoul_moving_mean,how='inner', on='도착 시군구 코드')
df_seoul_moving.head()

대상연월 요일  도착시간 출발 시군구 코드 도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분)_x  이동인구(합)_x  \
0  202204  일     5       노원구       성북구  F  65   EW             10       4.68   
1  202204  일     5       노원구       성북구  F  65   WH             10       9.44   
2  202204  일     5       노원구       성북구  F  70   HW             10       5.69   
3  202204  일     5       노원구       성북구  F  70   HE             10       5.68   
4  202204  일     5       노원구       성북구  F  70   EH             20       5.73   

   평균 이동 시간(분)_y    이동인구(합)_y  평균 이동 시간(분)  이동인구(합)  
0        5232500  11278789.10        22.72    48.97  
1        5232500  11278789.10        22.72    48.97  
2        5232500  11278789.10        22.72    48.97  
3        5232500  11278789.10        22.72    48.97  
4        5232500  11278789.10        22.72    48.97

In [88]:
seoul_moving.head()

대상연월 요일  도착시간  출발 시군구 코드  도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분)  이동인구(합)
0  202204  일     0      11010      11010  F   0   EE           10    28.02
1  202204  일     0      11010      11010  F   0   EH           10    27.52
2  202204  일     0      11010      11010  F   0   HE           10    27.83
3  202204  일     0      11010      11010  F  10   HW           10     6.05
4  202204  일     0      11010      11010  F  10   EH           90     5.95

In [89]:
df_seoul_moving.rename(columns={
    '평균 이동 시간(분)_y':'총 평균 이동 시간(분)',
    '이동인구(합)_y':'총 이동인구(합)',
    '평균 이동 시간(분)':'평균 이동 시간 평균(분)',
    '이동인구(합)':'이동인구(평균)',
    '평균 이동 시간(분)_x' : '평균 이동 시간(분)',
    '이동인구(합)_x': '이동인구(합)'}, inplace=True)

In [92]:
# 해당 데이터프레임을 'df_seoul_moving.csv' 파일로 저장하세요
df_seoul_moving.to_csv('df_seoul_moving.csv', index=False)

---

# 3.데이터 분석하기
+ KeyPoint : 데이터의 형태를 살펴보고 다양한 분석기법을 통해 모델링에 적합하도록 정제요소를 선별할 수 있다.
  * 데이터들의 패턴 탐색
  * 변수들간의 관계 파악

In [93]:
# 시각화 한글폰트 설정
import seaborn as sns
import matplotlib.pyplot as plt

plt.rc('font', family='nanum')
sns.set(font="NanumGothicCoding",#"NanumGothicCoding", 
        rc={"axes.unicode_minus":False}, # 마이너스 부호 깨짐 현상 해결
        style='darkgrid')

#### [실습문제10] 데이터 분포 알아보기
* 다양한 변수를 기준으로 그래프를 그려보고 인사이트를 도출해보세요.

In [ ]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.


In [ ]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.

In [ ]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.

In [ ]:
# 위 차트를 통해 알게된 사실을 정리해봅시다.
# 1. 
# 2.
# 3.

---